File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/BuyStageStartedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.LobbyState;

public class BuyStageStartedPayload {
    // Tells all the clients the buy stage has started.
    public LobbyState lobbyState;

    public BuyStageStartedPayload(LobbyState lobbyState) {
        this.lobbyState = lobbyState;
    }
}

File: front-end/src/dto/serverPayloads/BuyStageStartedPayload.js
import { LobbyState } from '../../models/LobbyState.js';

export class BuyStageStartedPayload {
    constructor(lobbyState) {
        this.lobbyState = lobbyState;
    }

    static fromDict(data) {
        return new BuyStageStartedPayload(LobbyState.fromDict(data.lobbyState));
    }
}
